src/net/server/middleware/processors/edns.rs

```diff
@@ -0,0 +1,294 @@
+//! RFC 6891 and related EDNS message processing.
+use core::ops::ControlFlow;
+
+use octseq::Octets;
+use tracing::{debug, trace, warn};
+
+use crate::base::iana::OptRcode;
+use crate::base::message_builder::AdditionalBuilder;
+use crate::base::opt::keepalive::IdleTimeout;
+use crate::base::opt::{Opt, OptRecord, TcpKeepalive};
+use crate::base::wire::Composer;
+use crate::base::{Message, StreamTarget};
+use crate::net::server::message::{Request, TransportSpecificContext};
+use crate::net::server::middleware::processor::MiddlewareProcessor;
+use crate::net::server::util::add_edns_options;
+use crate::net::server::util::start_reply;
+
+/// EDNS version 0.
+///
+/// Version 0 is the highest EDNS version number recoded in the [IANA
+/// registry] at the time of writing.
+///
+/// [IANA registry]: https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-14
+pub const EDNS_VERSION_ZERO: u8 = 0;
+
+/// A [`MiddlewareProcessor`] for adding EDNS(0) related functionality.
+///
+/// Standards covered by ths implementation:
+///
+/// | RFC    | Status  |
+/// |--------|---------|
+/// | [6891] | TBD     |
+/// | [7828] | TBD     |
+/// | [9210] | TBD     |
+///
+/// [6891]: https://datatracker.ietf.org/doc/html/rfc6891
+/// [7828]: https://datatracker.ietf.org/doc/html/rfc7828
+/// [9210]: https://datatracker.ietf.org/doc/html/rfc9210
+/// [`MiddlewareProcessor`]: crate::net::server::middleware::processor::MiddlewareProcessor
+#[derive(Debug)]
+pub struct EdnsMiddlewareProcessor {
+    /// Don't accept messages that advertize a higher EDNS version than this.
+    max_version: u8,
```

Comment:
Remove this until there is even an RFC that describes something other than zero.

Reply:
Done.